kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -74,6 +76,7 @@ public TransactionImpl(
         this.operation = operation;
         this.protocol = protocol;
         this.metadata = metadata;
+        this.setTxn = setTxn;
```

Comment:
can this be `setTxnOpt` ? makes it clearer that its an optional in other areas of the code